llvm/lib/Target/AMDGPU/AMDGPUInstructionSelector.cpp

```diff
@@ -120,6 +120,34 @@ bool AMDGPUInstructionSelector::constrainCopyLikeIntrin(MachineInstr &MI,
          RBI.constrainGenericRegister(Src.getReg(), *SrcRC, *MRI);
 }
 
+// Returns true if this is a copy from physical reg to VCC.
+// The parameter IsValid further indicates if the copy is valid.
+bool AMDGPUInstructionSelector::isCopyPhysicalToVCC(
+    Register DstReg, Register SrcReg, const MachineRegisterInfo &MRI,
+    bool &IsValid) const {
+  if (DstReg.isPhysical())
+    return false;
```

Comment:
In this case you should be taking the type from the virtual register operand. This should not just say no